### PR TITLE
Add scripts to fetch and generate protos

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

Many of the proto files used in the repo are copied from upstream gRPC
repos and built using the Makefile.  There's no easy wasy to update the
proto files and the Makefile has grown quite unwieldly and tends to be a
bit painful to update. It's also inconvenient to run each generate
target and targets often get missed.

Modifications:

- Add a 'Protos' directory containing two scripts: 'fetch.sh' and
  'generate.sh'.
- 'fetch.sh' clones a few upstream gRPC repos and selectively copies out
  protos we are interested in into 'Protos/upstream'. This makes it
  convenient to update all vendored protos.
- 'generate.sh' effectively replaces the Makefile with separaet
  functions for generating grpc code and messages.
- Move a few other protos from Sources/Tests into Protos
- Remove the Makefile

Result:

Much easier to update and re-generate protos.

### CODE REVIEW

**Comment by glbrntt** [comment]
Note this is in three commits:
- The scripts and a few repo changes in https://github.com/grpc/grpc-swift/pull/1784/commits/f78fd75975e500bc120769d8e4a2fda7429dd2c5
- Pulling in the upstream protos in [e029d79](https://github.com/grpc/grpc-swift/pull/1784/commits/e029d792394871085ebe864c81d5c5a42e3867ae)
- Re-generating in [bbe46ba](https://github.com/grpc/grpc-swift/pull/1784/commits/bbe46ba0775bba63dcb4a6dae06f28aadf79cb88)

File: Protos/upstream/grpc/reflection/v1alpha/reflection.proto [hunk 1]
```diff
@@ -11,13 +11,22 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 // Service exported by server reflection
 
+
+// Warning: this entire file is deprecated. Use this instead:
+// https://github.com/grpc/grpc-proto/blob/master/grpc/reflection/v1/reflection.proto
```
Comment (gjcairo) [inline Protos/upstream/grpc/reflection/v1alpha/reflection.proto:18]: Should we still be using this file (and all others that have been updated with this warning) at all?

File: Protos/generate.sh [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  generate_message "$proto" "$(dirname "$proto")" "$output" "Visibility=Public"
+  generate_grpc "$proto" "$(dirname "$proto")" "$output" "Visibility=Public"
+}
+
+function generate_reflection_service {
+  local proto_v1="$here/upstream/grpc/reflection/v1/reflection.proto"
+  local output_v1="$root/Sources/GRPCReflectionService/v1"
+
+  # Messages were accidentally leaked into public API, they shouldn't be but we
+  # can't undo that change until the next major version.
```
Comment (gjcairo) [inline Protos/generate.sh:104]: Are we tracking this?

File: Protos/upstream/grpc/reflection/v1alpha/reflection.proto (hunk 1 above)
Comment (glbrntt) [reply Protos/upstream/grpc/reflection/v1alpha/reflection.proto:18]: Yes, we offer both v1 and v1alpha, a number of tools using the reflection service only started using v1 recently. We also need to keep this for API compatibility reasons (our reflection service implementation can be configured to use  v1 or v1alpha).

File: Protos/generate.sh (hunk 2 above)
Comment (glbrntt) [reply Protos/generate.sh:104]: We are now: https://github.com/grpc/grpc-swift/issues/1785

**Review by gjcairo — APPROVED** [review]

**Pull request merged**